docs/source/quicktour.mdx

```diff
@@ -14,186 +14,188 @@ specific language governing permissions and limitations under the License.
 >>> for result in results:
-...     print(f"label: {result['label']}, with score: {round(result['score'], 4)}")
+...     print(f"label: {result["label"]}, with score: {round(result["score"], 4)}")
 label: POSITIVE, with score: 0.9998
 label: NEGATIVE, with score: 0.5309
 ```
 
-To use with a large dataset, look at [iterating over a pipeline](main_classes/pipelines)
+The [`pipeline`] can also iterate over an entire dataset. Start by installing the 🤗 [Datasets](https://huggingface.co/docs/datasets/) library:
+
+```bash
+pip install datasets 
+```
+
+Create a [`pipeline`] with the task you want to solve for and the model you want to use. Set the `device` parameter to `0` to place the tensors on a CUDA device:
+
+```py
+>>> from transformers import pipeline
+
+>>> speech_recognizer = pipeline("automatic-speech-recognition", model="facebook/wav2vec2-base-960h", device=0)
+```
 
-You can see the second sentence has been classified as negative (it needs to be positive or negative) but its score is
-fairly neutral.
+Next, load a dataset (see the 🤗 Datasets [Quick Start](https://huggingface.co/docs/datasets/quickstart.html#) for more details) you'd like to iterate over. For example, let's load the [SUPERB](https://huggingface.co/datasets/superb) dataset:
 
-By default, the model downloaded for this pipeline is called "distilbert-base-uncased-finetuned-sst-2-english". We can
-look at its [model page](https://huggingface.co/distilbert-base-uncased-finetuned-sst-2-english) to get more
-information about it. It uses the [DistilBERT architecture](model_doc/distilbert) and has been fine-tuned on a
-dataset called SST-2 for the sentiment analysis task.
+```py
+>>> import datasets
 
-Let's say we want to use another model; for instance, one that has been trained on French data. We can search through
-the [model hub](https://huggingface.co/models) that gathers models pretrained on a lot of data by research labs, but
-also community models (usually fine-tuned versions of those big models on a specific dataset). Applying the tags
-"French" and "text-classification" gives back a suggestion "nlptown/bert-base-multilingual-uncased-sentiment". Let's
-see how we can use it.
+>>> dataset = datasets.load_dataset("superb", name="asr", split="test")
+```
 
-You can directly pass the name of the model to use to [`pipeline`]:
+Now you can iterate over the dataset with the pipeline. `KeyDataset` retrieves the item in the dictionary returned by the dataset:
 
 ```py
->>> classifier = pipeline("sentiment-analysis", model="nlptown/bert-base-multilingual-uncased-sentiment")
+>>> for out in tqdm.tqdm(speech_recognizer(KeyDataset(dataset, "file"))):
+        print(out)
+        # {"text": "HE HOPED THERE WOULD BE STEW FOR DINNER TURNIPS AND CARROTS AND BRUISED POTATOES AND FAT MUTTON PIECES TO BE LADLED OUT IN THICK PEPPERED FLOWER FAT AND SAUCE"}
+        # {"text": ....}
+        # ....
```

Comment:
This shouldn't be set as comments here, we can use another code block to show the first output.